src/pyodide/internal/python.js

```diff
@@ -277,16 +277,45 @@ function simpleRunPython(emscriptenModule, code) {
   }
 }
 
-function mountLib(pyodide) {
+function buildSitePackages(tarInfo, requirements) {
+  const res = {
+    children: new Map(),
+    mode: 0o777,
+    type: 5,
+    modtime: 0,
+    size: 0,
+    path: "",
+    name: "",
+    parts: [],
+  };
+
+  for(const req of ["stdlib", ...requirements]) {
+    const child = tarInfo.children.get(req);
+    if(!child) {
+      throw new Error(`Requirement ${req} not found in pyodide packages tar`);
+    }
+    child.children.forEach((val, key) => {
+      if(res.children.has(key)) {
+        throw new Error(`Folder ${key} being written by multiple packages`);
+      }
+      res.children.set(key, val);
+    });
+  }
+
+  return res;
+}
+
+export function mountLib(pyodide, requirements) {
```

Comment:
Rather than exporting `mountLib` it looks like we could just call it at the start of `setupPackages`?

Reply:
setupPackages is in another file so we would still need to export it, but I agree it makes sense to call it from there.